pyface/ui/qt4/tasks/advanced_editor_area_pane.py

```diff
@@ -266,6 +293,11 @@ def __init__(self, editor_area, parent=None):
             QtCore.Qt.AllDockWidgetAreas, QtGui.QTabWidget.North
         )
 
+    def disconnect_event_listeners(self):
```

Comment:
I saw elsewhere there is this `_remove_event_listeners` which is defined in *other* interfaces, e.g.:
https://github.com/enthought/pyface/blob/f60905affb95cc0024c1a0514da4f4d389e5a533/pyface/i_widget.py#L89-L90

Looks like the intended effects would be the same, should we use the same name? We might want to declare this in the interface of `ITaskPane`.

Reply:
That interface is used in higher level objects, which would be `AdvancedEditorAreaPane` in this case. I chose a different name in downstream widgets on purpose. Also, the method has to be public so that it can be called from upstream.

Reply:
I think we have different levels of restrictions we want to impose: If the method does not have a preceding underscore, and the class name does not have a preceding underscore, and the module name does not have a preceding underscore, that is at the level of accessibility that we need to maintain backward compatibility for, and therefore it would impose limitations on how the code can change. Python is not like Java where there are more than one modifiers for imposing access control enforced by the compiler, so we have to resort to naming hints, comments etc.  I am not sure this `disconnect_event_listeners` is intended for consumption outside of pyface.

My main concern is the confusion this could cause to have two methods that have the same meaning but named differently.

Reply:
> My main concern is the confusion this could cause to have two methods that have the same meaning but named differently.

Another equally "main" concern is having a publicly named method that will be harder to change because we cannot be sure no one outside of pyface has used it.

Reply:
Changed the name of the method to `_remove_event_listeners`